src/com/facebook/buck/cli/TestCommand.java

```diff
@@ -605,6 +609,40 @@ public int compare(TestRule o1, TestRule o2) {
     return builder.build();
   }
 
+  @VisibleForTesting
+  Iterable<TestRule> filterUntestableRules(
+      Iterable<TestRule> testRules) {
+
+    ImmutableSortedSet.Builder<TestRule> builder =
+        ImmutableSortedSet.orderedBy(
+            new Comparator<TestRule>() {
+              @Override
+              public int compare(TestRule o1, TestRule o2) {
+                return o1.getBuildTarget().getFullyQualifiedName().compareTo(
+                    o2.getBuildTarget().getFullyQualifiedName());
+              }
+            });
+
+    for (TestRule rule : testRules) {
+      // xctool doesn't run XCUITests yet, but project generation works
+      boolean isAppleUITest = false;
+      if (rule instanceof AppleTest) {
+        AppleTest appleTestRule = (AppleTest)rule;
+        isAppleUITest = appleTestRule.isUiTest();
+        if (isAppleUITest) {
+          LOG.warn("Not running rule '%s' as XCUITests aren't supported.",
+              rule.getFullyQualifiedName());
+        }
+      }
+
+      if (!isAppleUITest) {
+        builder.add(rule);
+      }
+    }
+
+    return builder.build();
+  }
```

Comment:
Should you let AppleTest decide if it can run UI tests or not? There is a `useXctest` boolean flag. I wonder, if it is set to true, can it invoke ui tests using `xctest`?